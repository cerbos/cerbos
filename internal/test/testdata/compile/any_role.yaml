---
mainDef: "resource_policies/leave_request_20210210.yaml"
inputDefs:
  "resource_policies/leave_request_20210210.yaml":
    apiVersion: api.cerbos.dev/v1
    resourcePolicy:
      resource: leave_request
      version: "20210210"
      importDerivedRoles:
        - my_derived_roles
      schemas:
        principalSchema:
          ref: cerbos:///complex_object.json
        resourceSchema:
          ref: cerbos:///complex_object.json
      rules:
        - actions: ['*']
          effect: EFFECT_ALLOW
          roles:
            - admin
            - manager
            - '*'
          name: wildcard
        - actions: ["create"]
          derivedRoles:
            - tester
          effect: EFFECT_ALLOW
 

  "derived_roles/my_derived_roles.yaml":
    apiVersion: "api.cerbos.dev/v1"
    derivedRoles:
      name: my_derived_roles
      definitions:
        - name: tester
          parentRoles: ["dev", "qa", "*"]

