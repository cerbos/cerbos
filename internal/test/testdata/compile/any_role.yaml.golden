{
  "fqn": "cerbos.resource.leave_request.v20210210",
  "resourcePolicy": {
    "meta": {
      "fqn": "cerbos.resource.leave_request.v20210210",
      "resource": "leave_request",
      "version": "20210210",
      "sourceAttributes": {
        "resource.leave_request.v20210210": {
          "attributes": {
            "source": "resource_policies/leave_request_20210210.yaml"
          }
        }
      }
    },
    "policies": [
      {
        "derivedRoles": {
          "tester": {
            "name": "tester",
            "parentRoles": {
              "*": {}
            }
          }
        },
        "rules": [
          {
            "name": "wildcard",
            "actions": {
              "*": {}
            },
            "roles": {
              "*": {}
            },
            "effect": "EFFECT_ALLOW"
          },
          {
            "name": "rule-002",
            "actions": {
              "create": {}
            },
            "derivedRoles": {
              "tester": {}
            },
            "effect": "EFFECT_ALLOW"
          }
        ],
        "schemas": {
          "principalSchema": {
            "ref": "cerbos:///complex_object.json"
          },
          "resourceSchema": {
            "ref": "cerbos:///complex_object.json"
          }
        }
      }
    ],
    "schemas": {
      "principalSchema": {
        "ref": "cerbos:///complex_object.json"
      },
      "resourceSchema": {
        "ref": "cerbos:///complex_object.json"
      }
    }
  },
  "compilerVersion": 1
}
